JS/index.js

```diff
@@ -0,0 +1,184 @@
+console.log('hello');
+let countries = [];
+const containerCountries = document.querySelector(".countries-grid");
+const countryCardTemplate = document.querySelector(".country.scale-effect");
+
+fetch('./CountriesData.json')
+    .then((res) => {
+        if (!res.ok) {
+            throw new Error(`HTTP error! Status: ${res.status}`);
+        }
+        return res.json();
+    })
+    .then((data) => {
+        countries = data;
+        console.log(data);
+        innerDetailsIntoCards(countries); 
+    })
+    .catch((error) => {
+        console.error("Unable to fetch data:", error);
+    });
+
+const innerDetailsIntoCards = (array) => {
+    containerCountries.innerHTML = '';
+
+    array.forEach((country) => {
+        const cloneCard = countryCardTemplate.cloneNode(true);
+       
+        const titleElement = cloneCard.querySelector(".country-title");
+        const flagImage = cloneCard.querySelector(".country-flag img");
+        const countryInfo = cloneCard.querySelectorAll(".country-brief li");
+
+        titleElement.innerHTML = country.name;
+        flagImage.src = country.flag;
+        flagImage.alt = `${country.name}`;
+        
+        countryInfo[0].innerHTML = `<strong>Population:</strong> ${country.population}`;
+        countryInfo[1].innerHTML = `<strong>Region:</strong> ${country.region}`;
+        countryInfo[2].innerHTML = `<strong>Capital:</strong> ${country.capital}`;
+
+        containerCountries.appendChild(cloneCard);
+    });
+};
+
+const openDropDown = () => {
+  const dropDown = document.querySelector(".dropdown-wrapper");
+  console.log('enter');
+  dropDown.classList.toggle("open");
+}
+
+const filterByRegion = (event) => {
+   const region = event.target.getAttribute("data-region");
+   console.log("select region:" + region)
+
+   if(region!="all"){
+   const result = countries.filter((country)=>{
+    return country.region.toLowerCase() == region.toLowerCase()
+   })
+   console.log(result);
+   innerDetailsIntoCards(result);
+}
+else{
+    innerDetailsIntoCards(countries);
+}
+
+//    const result =  countries.filter((country)=>{
+//          return country.region === "Africa";
+//     })
+//     console.log(result);
+}
+
+const matchValues = (country, input) => {
+    return (country.name.toLowerCase().includes(input) || 
+        country.population.toString().includes(input)) ||
+        country.region.toLowerCase().includes(input) ||
+        country.capital.toLowerCase().includes(input)
+}
```

Comment:
NICE! 

Consider a case where the user enters " Israel   " (with spaces and tabs before and after the word)
https://www.geeksforgeeks.org/string-strip-in-javascript/